parts/kubernetesagentcustomdataforcoreos.yml

```diff
@@ -0,0 +1,120 @@
+  permissions: "0644"
+  encoding: "base64"
+  owner: "root"
+  content: |
+    {{WrapAsVariable "clientCertificate"}}
+
+- path: "/var/lib/kubelet/kubeconfig"
+  permissions: "0644"
+  owner: "root"
+  content: |
+    apiVersion: v1
+    kind: Config
+    clusters:
+    - name: localcluster
+      cluster:
+        certificate-authority: /etc/kubernetes/certs/ca.crt
+        server: https://{{WrapAsVariable "kubernetesAPIServerIP"}}:443
+    users:
+    - name: client
+      user:
+        client-certificate: /etc/kubernetes/certs/client.crt
+        client-key: /etc/kubernetes/certs/client.key
+    contexts:
+    - context:
+        cluster: localcluster
+        user: client
+      name: localclustercontext
+    current-context: localclustercontext
+
+- path: "/etc/systemd/system/kubectl-extract.service"
+  permissions: "0644"
+  owner: "root"
+  content: |
+    [Unit]
+    Description=Kubectl extraction
+    Requires=docker.service
+    After=docker.service
+    ConditionPathExists=!/opt/kubectl
+
+    [Service]
+    TimeoutStartSec=0
+    Restart=on-failure
+    RestartSec=5s
+    ExecStartPre=/bin/mkdir -p /tmp/kubectldir
+    ExecStartPre=/usr/bin/docker pull {{WrapAsVariable "kubernetesHyperkubeSpec"}}
+    ExecStartPre=/usr/bin/docker run --rm -v /tmp/kubectldir:/opt/kubectldir {{WrapAsVariable "kubernetesHyperkubeSpec"}} /bin/bash -c "cp /hyperkube /opt/kubectldir/"
+    ExecStartPre=/bin/mv /tmp/kubectldir/hyperkube /opt/kubectl
+    ExecStart=/bin/chmod a+x /opt/kubectl
+
+    [Install]
+    WantedBy=multi-user.target
+
+- path: "/etc/default/kubelet"
+  permissions: "0644"
+  owner: "root"
+  content: |
+    KUBELET_CLUSTER_DNS={{WrapAsVariable "kubeDnsServiceIP"}}
+    KUBELET_API_SERVERS=https://{{WrapAsVariable "kubernetesAPIServerIP"}}:443
+    KUBELET_IMAGE={{WrapAsVariable "kubernetesHyperkubeSpec"}}
+    KUBELET_NETWORK_PLUGIN=kubenet
+    DOCKER_OPTS=
+    CUSTOM_CMD=/bin/true
+    KUBELET_REGISTER_SCHEDULABLE=true
+    KUBELET_NODE_LABELS={{ GetKubernetesLabels . }}
+    KUBELET_POD_INFRA_CONTAINER_IMAGE={{WrapAsVariable "kubernetesPodInfraContainerSpec"}}
+    KUBELET_NODE_STATUS_UPDATE_FREQUENCY={{WrapAsVariable "kubernetesNodeStatusUpdateFrequency"}}
+    KUBE_CTRL_MGR_NODE_MONITOR_GRACE_PERIOD={{WrapAsVariable "kubernetesCtrlMgrNodeMonitorGracePeriod"}}
+    KUBE_CTRL_MGR_POD_EVICTION_TIMEOUT={{WrapAsVariable "kubernetesCtrlMgrPodEvictionTimeout"}}
+    KUBE_CTRL_MGR_ROUTE_RECONCILIATION_PERIOD={{WrapAsVariable "kubernetesCtrlMgrRouteReconciliationPeriod"}}
+{{if IsKubernetesVersionGe "1.6.0"}}
+    KUBELET_FEATURE_GATES=--feature-gates=Accelerators=true
+{{end}}
+
+- path: "/etc/systemd/system/kubelet.service"
+  permissions: "0644"
+  encoding: gzip
+  owner: "root"
+  content: !!binary |
+    KUBELET_SERVICE_B64_GZIP_STR
+
+- path: "/opt/azure/containers/kubelet.sh"
+  permissions: "0755"
+  owner: "root"
+  content: |
+    #!/bin/bash
+    exit 0
+
+- path: "/opt/azure/containers/provision.sh"
```

Comment:
One of the differences I see between CoreOS agent custom data vs Ubuntu is absence of a runcmd. Is this because CoreOS comes installed with Docker Engine and other packages?